src/services/rfqm_service.ts

```diff
@@ -439,6 +450,118 @@ export class RfqmService {
         return Math.random() > 0.5 ? Promise.reject('failing') : Promise.resolve();
     }
 
+    /**
+     * Submit a transaction, re-submit if necessary, and poll for resolution
+     */
+    public async completeSubmissionLifecycleAsync(
+        orderHash: string,
+        workerAddress: string,
+        callData: string,
+    ): Promise<void> {
+        const submissions: { [transactionHash: string]: RfqmTransactionSubmissionEntity } = {};
+
+        let gasPrice = await this._protocolFeeUtils.getGasPriceEstimationOrThrowAsync();
+
+        const firstSubmission = await this._submitTransactionAsync(orderHash, workerAddress, callData, gasPrice);
+        submissions[firstSubmission.transactionHash!] = firstSubmission;
+
+        let isFinishedMonitoring = false;
+        let shouldTryResubmitting = true;
+        while (!isFinishedMonitoring) {
+            await sleepAsync(TRANSACTION_WATCHER_SLEEP_TIME_MS);
```

Comment:
I think we have a library called `delay`.

`await delay(TRANSACTION_WATCHER_SLEEP_TIME_MS`

Reply:
👍